pkg/crosscluster/logical/replication_statements.go

```diff
@@ -153,31 +181,37 @@ func newMatchesLastRow(columns []catalog.Column, startParamIdx int) (tree.Expr,
 func newUpdateStatement(
 	table catalog.TableDescriptor,
 ) (statements.Statement[tree.Statement], error) {
-	columns := getPhysicalColumns(table)
+	columns := getColumnSchema(table)
 
 	// Create WHERE clause for matching the previous row values
 	whereClause, err := newMatchesLastRow(columns, 1)
 	if err != nil {
 		return statements.Statement[tree.Statement]{}, err
 	}
 
-	exprs := make(tree.UpdateExprs, len(columns))
+	exprs := make(tree.UpdateExprs, 0, len(columns))
 	for i, col := range columns {
-		nameNode := tree.Name(col.GetName())
+		if col.isComputed {
+			// Skip computed columns since they are not needed to fully specify the
+			// contents of the row.
```

Comment:
i hope @michae2 's fast path handles stored computed column that may be apart of the pk!

Reply:
Sigh... looks like I still have some work to do.

Reply:
Also, it looks like my optimizer rules aren't recognizing the generic query plan, only the custom query plan. So it's possible this never gets the fast path, due to the generic query plan. I will fix that, too.